### ISSUE TITLE
TS lets npm package types affect the compilation errors (not respecting target)

### ISSUE THREAD

**Issue by cscleison** [microsoft/TypeScript#40388]
Sorry I can't find a good title for this.

**TypeScript Version:** noticed in 3.8, and tested also 3.9.7 and 4.1.0-dev.20200904

**Search Terms:**
typescript npm package types affecting compilation
typescript package types don't respect target

**Code**

Repository with a [repro](https://github.com/cscleison/ts-error) of the problem 

`tsconfig.json`
```
{
  "compilerOptions": {
    "strict": true,
    "allowJs": false,
    "target": "es5"
  },
  "include": ["./src"]
}
```

`src/index.ts`
```ts  
// should have no errors
["a", "b"].map((c) => c.toUpperCase());

// TS should complain about `includes` because it's not part of ES5
[1, 2, 3].includes(0);
```

**What is the problem**
If I just have the `typescript` package installed, I get a compilation error **as expected** for the `.includes` usage.
However when I install `npm i -D webpack-dev-server` or `npm i -D @types/enzyme` the compilation error is silently gone - not respecting what I've configured in `target`.

These packages reuse Nodejs types, and those internally use new ES specs types. Even if I don't import anything, TypeScript loads and uses them. I'm not sure if this is a known issue or if this is supposed to happen. Or who's the real culprit here (the libraries or TS). Is there any workaround to avoid this? Seems a bit dangerous to let an unsupported ES feature (for my targets) slip in. Also it's quite tricky to find who's the lib causing this.

**Expected behavior:**
Libraries internal types should not affect the configuration I've made in `target`

**Actual behavior:**
Installing 3rd party libs affect the libs/target and can silently allow unsupported ES features.

**Comment by ilogico** [comment]
You can restrict the automatically included [types(https://www.typescriptlang.org/tsconfig#types).

**Comment by cscleison** [comment]
To me it seems that the libs should not be referencing node. The problem is that there are **a lot of libs** doing it.

**Comment by cscleison** [comment]
i'm closing this and will try to reach out the libs definitions first 
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/47341

**Comment by cscleison** [comment]
thanks @ilogico 
`types: []` did the trick
